### DISCUSSION TITLE
Various 'contexts' within ApplicationContext

### DISCUSSION THREAD

**Discussion by Chicagoan2016** [MarimerLLC/csla#3424] [Questions] [answered]
I have been asked to keep track of User's office location for audit purposes.
We have office id for each employee available in the security database, but some employees are 'roaming' employees. 
I thought of storing their office ids in one of the context dictionaries within ApplicationContext.
Is it a recommended practice? if so, which one should I use? global, local, or client?

Kind Regards

**Comment by rockfordlhotka** [comment, marked as answer]
Given that this data element is per-user, I'd strongly consider putting it as a claim in the ClaimsIdentity for the user.

If you don't want to do that, then probably putting it in `ClientContext` makes the most sense, so it flows with data portal calls.

There is no `GlobalContext` anymore - that has been removed.

**Reply by Chicagoan2016** [reply]
@rockfordlhotka Thank you Sir, it's an older application from years ago (Csla 4.xx). 
So I guess ClientContext it is. If we have  officeid property in  CslaIdentity, we couldn't write that property for 'roaming' users?

Roaming users will be able to switch to different offices after login.
